corgie/cli/align.py

```diff
@@ -220,54 +234,198 @@ def align(
+        block_field_name,
+        layer_type="field",
+        overwrite=False,
+    )
+    odd_stack.create_sublayer(
+        block_field_name,
+        layer_type="field",
+        overwrite=False,
+    )
+    composed_field = dst_stack.create_sublayer(
+        f"composed_field{suffix}", layer_type="field", overwrite=True
+    )
+    if (restart_stage > 2) and (stitch_size > 1):
+        stitch_corrected_field = dst_stack.create_sublayer(
+            f"stitch_corrected{suffix}", layer_type="field", overwrite=False
         )
-
-    scheduler.execute_until_completion()
-    corgie_logger.debug("Done!")
-
-    corgie_logger.debug("Stitching blocks...")
-    # TODO
-    # stitch_blocks_job = StitchBlockJob(
-    #    blocks=blocks,
-    #    src_stack=src_stack,
-    #    dst_stack=dst_stack,
-    #    bcube=bcube,
-    #    suffix=suffix,
-    #    render_method=render_method,
-    #    cf_method=cf_method
-    # )
-
-    # scheduler.register_job(stitch_blocks_job, job_name=f"Stitch blocks {bcube}")
-    # scheduler.execute_until_completion()
-
-    corgie_logger.debug("Done!")
+    if restart_stage <= 3:
+        corgie_logger.debug("Stitching blocks...")
+        for block, stitch_block in zip(blocks[1:], stitch_blocks):
+            block_bcube = block.get_bcube(bcube)
+            block_list = block.get_neighbors(dist=decay_dist)
+            corgie_logger.debug(f"src_block: {block}")
+            corgie_logger.debug(f"influencing blocks: {block_list}")
+            z_list = [b.stop for b in block_list]
+            # stitch_corrected_field used if there is multi-section block overlap,
+            # which requires voting to produce a corrected field.
+            # If there is only single-section block overlap, then use
+            # stitch_estimated_fields from each stitch_block
+            if stitch_corrected_field is not None:
+                stitching_fields = [stitch_corrected_field]
+            else:
+                # Order with furthest block first (convention of FieldSet).
+                stitching_fields = [
+                    stitch_block.dst_stack[stitch_estimated_name],
+                    stitch_block.src_stack[stitch_estimated_name],
+                ]
+
+            broadcast_job = BroadcastJob(
+                block_field=block.dst_stack[block_field_name],
+                stitching_fields=stitching_fields,
+                output_field=composed_field,
+                chunk_xy=chunk_xy,
+                bcube=block_bcube,
+                pad=pad,
+                z_list=z_list,
+                mip=processor_mip[0],
+                decay_dist=decay_dist,
+            )
+            scheduler.register_job(
+                broadcast_job, job_name=f"Broadcast {block} {block_bcube}"
+            )
+
+        scheduler.execute_until_completion()
+        corgie_logger.debug("Done!")
+
+    # TODO: copy first block's field to composed_field layer
+
+    if restart_stage <= 4:
+        for block in blocks[:1]:
```

Comment:
`block = blocks[0]` is more clear

Reply:
sorry I don't have the full context but if `blocks` can be an empty list `for block in blocks[:1]:` won't throw an error, `blocks[0]` will result in index out of range